requirements.txt

```diff
@@ -3,3 +3,4 @@ openai==0.27.4
 chromadb==0.3.21
 pre-commit>=3.2.0
 python-dotenv==1.0.0
+llama-cpp-python==0.1.36
```

Comment:
Problem with laving llama in the main requirements is that it's now installed for anyone, including people that run on micro instances in EC2, and it pulls CUDA, torch, and tons of other stuff... it's now in extensions/requirements.txt.

Reply:
That is incorrect.
llama_cpp does not pull cuda or torch.
chromadb is the insanely heavy import that pulls gigabytes of cuda, torch and so on.
llama_cpp is only a couple of hundred kilobytes, see the wheels here https://github.com/abetlen/llama-cpp-python/releases and has no weighty dependencies

Reply:
Ha, for whatever reason I thought llama was the weighty one... Hm, now I need to figure out how to deal with Chroma.

I still think llama-cpp should be in extensions/requirements.txt mostly because it's not part of the default behavior, and we want people to be able to grab just babyagi.py and requirements.txt and have the minimal working set.

Reply:
Hmm, I think download weight is overrated nowadays.
For me with a sub-par internet connection it takes 390s to build the docker image.
People are happily pulling 30 billion parameter models just to try out another flavour.
It all really doesn't matter as long as it just works :) What really is taking time (=annoying) to build the image 5 times to figure put what's all missing. Creating issues and pull requests ;)

Reply:
To be fair I am in the same boat. But as I mentioned, there are folks who are trying to run on micro EC2 instances, where the size of the download is rather important, and pulling stuff they don't need can actually break them.